main.py

```diff
@@ -0,0 +1,420 @@
+        # Look through the movie library to build an index of the known movies
+        movies = []
+        for library_entry in self.library.movies.values():
+            # If one or more genre_id values were provided in the url then only include movies with that genre_id
+            if 'genre_id' in self.parsed_qs:
+                allowed_genre_ids = [int(genre_id) for genre_id in self.parsed_qs['genre_id']]
+                movie_genre_ids = library_entry['tmdb_details']['genre_ids']
+                for genre_id in movie_genre_ids:
+                    if genre_id in allowed_genre_ids:
+                        break
+                else:
+                    continue
+
+            # If one or more original_language values were provided in the url then only include movies with that original_language
+            if 'original_language' in self.parsed_qs:
+                allowed_original_languages = self.parsed_qs['original_language']
+                if library_entry['tmdb_details']['original_language'] not in allowed_original_languages:
+                    continue
+
+            # If all specified filters matched then include this movie in the index
+            movies.append(library_entry['tmdb_details'])
+
+        # Send the movie index json to the client
+        f = io.BytesIO(json.dumps(movies).encode('utf-8'))
+        self.send_fileobj(f, mime_type='application/json')
+
+    def send_movie_mkv(self, movie_id):
+        if movie_id not in self.library.movies:
+            return self.send_error(HTTPStatus.NOT_FOUND)
+
+        filename = movie_id.split('/')[1] + '.mkv'
+
+        path = self.library.movies[movie_id]['path']
+
+        with path.open(mode='rb') as f:
+            self.send_fileobj(f, mime_type='video/x-matroska')
+
+    def send_movie_attachment(self, movie_id, attachment_name):
+        # Find the specified mkv
+        if movie_id not in self.library.movies:
+            return self.send_error(HTTPStatus.NOT_FOUND)
+        path = self.library.movies[movie_id]['path']
+
+        # Get the list of attachments on this mkv
+        identify_result = subprocess.run(['mkvmerge', '--identify', path.as_posix()], capture_output=True)
+        if identify_result.returncode != 0:
+            return self.send_error(HTTPStatus.INTERNAL_SERVER_ERROR)
+
+        # Find the attachment by name
+        for line in identify_result.stdout.decode('utf-8').split('\n'):
+            m = re.match(r"Attachment ID (?P<attachment_id>[0-9]+): type '(?P<mime_type>[^']+)',(.*,?) file name '(?P<filename>[^']+)'.*", line)
+            if m and m.group('filename') == attachment_name:
+                break
+        else:
+            return self.send_error(HTTPStatus.NOT_FOUND)
+
+        # Extract the pertinent information from the metadata line
+        mime_type = m.group('mime_type')
+        attachment_id = m.group('attachment_id')
+        filename = m.group('filename')
+
+        # Extact the attachment content
```

Comment:
There is a typo in the comment.

Reply:
https://github.com/duckflix/media-server/issues/5